tests/test_core.py

```diff
@@ -7806,6 +7806,7 @@ def test_async_ccall_promise(self, exit_runtime):
     self.set_setting('INVOKE_RUN', 0)
     self.set_setting('EXIT_RUNTIME', exit_runtime)
     self.set_setting('EXPORTED_FUNCTIONS', ['_stringf', '_floatf'])
+    self.set_setting('DEFAULT_LIBRARY_FUNCS_TO_INCLUDE', ['$maybeExit'])
```

Comment:
Yes... the bug here is that we were serializing False as the string "False" and generating `-sEXIT_RUNTIME=False` on the command line which emcc was interpreting as truthy.  i.e. `-sEXIT_RUNTIME=False` is the the same as `-sEXIT_RUNTIME=1`

This change just fixes what the tests pass.

I've got a larger followup change that adds better type checking to emcc to make this into an error: #16539

Reply:
The value of `exit_runtime` in this function is a python boolean false. Which means we were previously building with `-sEXIT_RUNTIME=False` instead of `-sEXIT_RUNTIME=0` .. which means we were never actually testing the EXIT_RUNTIME=0 case.. and in that case the `'$maybeExit` (which we use in this test) was not getting included (since its not otherwise used).

Reply:
I see, thanks... that was deeper than I'd expected.